lib/QC_Dir.qpp

```diff
@@ -65,6 +65,7 @@ QoreListNode* qore_qd_private::list(ExceptionSink *xsink, int stat_filter, const
     ON_BLOCK_EXIT(closedir, dptr);
 
     struct dirent *de;
+    ExceptionSink tempSink;
```

Comment:
instead of using an ExceptionSink object, just call `::stat()` normally - the `ExceptionSink` provides no value - the `stat()` function in `qore_qd_private` is only a thin wrapper around `stat()` that throws a Qore-language exception in the `ExceptionSink` object